src/oclif/hooks/init.ts

```diff
@@ -188,6 +203,81 @@ export default async function(options: HookKeyOrOptions<'init'>) {
     }
   }
 
+  help.prototype.showHelp = function(_argv: string[]) {
+    const subject = getHelpSubject(_argv)
+    if (subject) {
+      const command = this.config.findCommand(subject)
+      if (command) {
+        this.showCommandHelp(command)
+        return
+      }
+
+      const topic = this.config.findTopic(subject)
+      if (topic) {
+        this.showTopicHelp(topic)
+        return
+      }
+
+      error(`command ${subject} not found`)
+    }
+
+    const renderCommands = (commands: any): string => {
+      return renderList(
+        commands.map(c => [chalk.hex(COLORS.PINK)(c.name), c.description && this.render(c.description.split('\n')[0])]),
+        {
+          spacer: '\n',
+          stripAnsi: this.opts.stripAnsi,
+          maxWidth: this.opts.maxWidth - 2,
+        }
+      )
+    }
+
+    const commandsGroup: Record<string, number> = FeatureFlag.getSingleton().getFeatureFlagInfo<Record<string, number>>(
+      'COMMANDS_GROUP'
+    )
+    const commandsId: Record<number, string> = FeatureFlag.getSingleton().getFeatureFlagInfo<Record<number, string>>(
+      'COMMANDS_GROUP_ID'
+    )
+    const commandsGroupLength: number = Object.keys(commandsId).length
+
+    const commands = this.config.commands
+      .filter(c => !c.id.includes(':'))
+      .map(c => {
+        return { name: c.id, description: c.description }
+      })
+    const topics = this.config.topics
+      .filter(t => !t.name.includes(':'))
+      .map(c => {
+        return { name: c.name, description: c.description }
+      })
+    const allCommands = commands.concat(topics)
+
+    const groups = []
+    for (let i = 0; i < commandsGroupLength; i++) groups.push([])
+
+    const cachedObject: Map<string, boolean> = new Map<string, boolean>()
+    for (let i = 0; i < allCommands.length; i++) {
+      const command: CommandI = allCommands[i]
+      if (cachedObject.has(command.name)) continue
+      cachedObject.set(command.name, true)
+
+      const commandGroupId = commandsGroup[command.name]
+
+      if (commandGroupId) {
+        groups[commandGroupId].push(command)
+      } else {
+        groups[commandsGroupLength - 1].push(command)
+      }
+    }
```

Comment:
nit:
A suggestion to make the code a little bit more functional style.
```suggestion
    const groups = Object.keys(commandsId).map(_ => [])
    
    const cachedObject: Map<string, boolean> = new Map<string, boolean>()

    allCommands.forEach((command: CommandI) => {
      if (cachedObject.has(command.name)) return
      cachedObject.set(command.name, true)

      const commandGroupId = commandsGroup[command.name]

      if (commandGroupId) {
        groups[commandGroupId].push(command)
      } else {
        groups[commandsGroupLength - 1].push(command)
      }
    })
```